parlai/core/utils.py

```diff
@@ -663,6 +678,17 @@ def build():
                     if mod_p not in self.white_list:
                         self.add_phrase(mod_p)
 
+    def memo(f):
```

Comment:
you can replace this with [functools.lru_cache](https://docs.python.org/3/library/functools.html#functools.lru_cache)